### PR TITLE
GLV implementation for BLS12_377, BLS12_381, BN254, BW6_761, Bandersnatch, Pallas and Vesta

### PR DESCRIPTION
**Pull request by simonmasson**
## Description

Implementaiton of the GLV method for several curves.

This PR is binded to [this PR]https://github.com/arkworks-rs/algebra/pull/588)  done in the `algebra` repository. The benchmarks should be moved into `algebra` but I did not find the way to write it.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by simonmasson** [comment]
@Pratyush @weikengchen @zhenfeizhang

File: bls12_377/Cargo.toml [hunk 1]
```diff
@@ -12,17 +12,7 @@ include = ["Cargo.toml", "src", "README.md", "LICENSE-APACHE", "LICENSE-MIT"]
...
-ark-ff = { version= "0.4.0", default-features = false }
-ark-ec = { version= "0.4.0", default-features = false }
-ark-r1cs-std = { version= "0.4.0", default-features = false, optional = true }
-ark-std = { version = "0.4.0", default-features = false }
-
 [dev-dependencies]
-ark-relations = { version= "0.4.0", default-features = false }
-ark-serialize = { version = "0.4.0", default-features = false }
-ark-algebra-test-templates = { version = "0.4.0", default-features = false }
-ark-algebra-bench-templates = { version = "0.4.0", default-features = false }
```
Comment (Pratyush) [inline bls12_377/Cargo.toml:25]: Why were these removed?
